articles/feature-engineering-in-machine-learning/index.md

```diff
@@ -0,0 +1,167 @@
+
+- **Choosing a model**- this step involves selection of a model best suited to the prepared data.
+
+- **Training**-  the model is trained in this stage (using data). Data may be split as test data and training data. Test data is used to test the model. Training data trains the model.
+
+- **Evaluation**- the model performance is evaluated at this stage through a number of metrics.
+
+- **Model deployment**- the models are deployed. They are integrated into their intended processes or applications.
+
+- **Performance monitoring**- a model should always be monitored after deployment. This ensures the model works correctly. It also provides the chance to update models or carry out maintenance on them.
+
+### Feature Engineering
+As mentioned previously, feature engineering falls into the data preparation stage of the machine learning process. Let us finally define feature engineering. Feature engineering is the process of creating new input features for machine learning.
+
+Features are extracted from raw data. These features are then transformed into formats compatible with the machine learning process. Domain knowledge of data is key to the process. Along with domain knowledge, both programming and math skills are required to perform feature engineering. The correct features make machine learning algorithms successful.
+
+This is one of the most effective ways to improve model performance. But what is a feature? According to Wikipedia, a [feature](https://en.wikipedia.org/wiki/Feature_engineering) is a property shared by all the independent units on which analysis is to be done.
+
+It is a characteristic that might assist in problem solving. As beneficial as this process is, it is often described as being time consuming. It also is a difficult process. Furthermore, it requires domain expertise for the data being analyzed. So, why exactly do we need this process?
+
+#### Need for Feature Engineering
+Features influence results of predictive models. I'd like to highlight two points on the importance of this process:
+
+1. **Reduced complexity**: Algorithms are fed raw data in order to build these models. However, these algorithms make predictions without a clear guide. Feature engineering guide these algorithms. But what is the benefit of this? For starters, with correct features, model complexity will reduce. Correct scope and purpose of the model makes the process more efficient. This makes it simpler to understand, build, modify, and maintain models.
+
+2. **Increased accuracy**: Feature engineering is a broad process. It involves transforming variables into suitable formats. It involves creation of new variables from existing ones. Filling in missing values, among other processes, is also under this umbrella. These processes ultimately influence accuracy of the model. When done correctly, with the correct data, it increases the accuracy of the models.
```

Comment:
Consider giving an example of suitable formats that you are referring to. I want to know what variables are and under what circumstances the need to change their formats arises. What types of variables usually go through these changes? Categorical ones, numerical ones?